firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md

```diff
@@ -0,0 +1,374 @@
+The _Export Collections to BigQuery_ extension also provides some BigQuery
+[user-defined functions](https://github.com/firebase/extensions/blob/master/firestore-bigquery-export/scripts/gen-schema-view/src/udf.ts)
+that are helpful in converting Firestore document properties to richly-typed
+BigQuery cells.
+
+## Use the script
+
+The following steps are an example of how to use the schema-views script. In the
+sections at the end of this file, you can review detailed information about
+configuring a schema file and reviewing the resulting schema views.
+
+### Step 1: Create a schema file
+
+The schema-views script runs against "schema files" which specify your schema
+configurations in a JSON format.  
+  
+In any directory, create a schema file called `test_schema.json`
+that contains the following:
+
+```
+{
+  "fields": [
+    {
+      "name": "name",
+      "type": "string"
+    },
+    {
+      "name": "age",
+      "type": "number"
+    }
+  ]
+}
+```
+
+Learn [How to configure schema files](#how-to-configure-schema-files)
+later in this guide.
+
+### Step 2: Set up credentials
+
+The schema-views script uses Application Default Credentials to communicate with
+BigQuery. 
+
+One way to set up these credentials is to run the following command using the
+[gcloud](https://cloud.google.com/sdk/gcloud/) CLI:
+
+```
+$ gcloud auth application-default login
+```
+
+Alternatively, you can
+[create and use a service account](https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually).
+This service account must be assigned a role that grants the permission of
+`bigquery.jobs.create`.
+
+### Step 3: Run the script
+
+The schema-views script uses the following parameter values from your
+installation of the extension:
+
++   `${param:PROJECT_ID}`: the project ID for the Firebase project in
+    which you installed the extension
++   `${param:DATASET_ID}`: the ID that you specified for your dataset during
+    extension installation
++   `${param:TABLE_ID}`: the common prefix of BigQuery views to generate
+
+Run the schema-views script using
+[`npx` (the Node Package Runner)](https://github.com/npm/npx#npx1----execute-npm-package-binaries)
+via `npm` (the Node Package Manager).
+
+1.  Make sure that you've installed the required tools to run the
+    schema-views script:
+    +   To access the `npm` command tools, you need to install
+        [Node.js](https://www.nodejs.org/).
+    +   If you use npm v5.1 or earlier, you need to explicitly install `npx`.
+        Run `npm install --global npx`.
+
+1.  Run the schema-views script via `npx` by running the following command:
+
+    ```
+    $ npx @firebaseextensions/fs-bq-schema-views \
+      --non-interactive
+      --project=${param:PROJECT_ID}
+      --dataset=${param:DATASET_ID}
+      --table-name-prefix=${param:TABLE_ID}
+      --schema-files=./test_schema.json
```

Comment:
To make this copy and pasteable, it needs a `\` before every newline.

```markdown
```shell
npx @firebaseextensions/fs-bq-schema-views \
      --non-interactive \
      --project=${param:PROJECT_ID} \
      --dataset=${param:DATASET_ID} \
      --table-name-prefix=${param:TABLE_ID} \
      --schema-files=./test_schema.json
```

Reply:
done :-)